src/protocol/context/malicious.rs

```diff
@@ -96,28 +89,71 @@ impl<'a, F: Field> Context<F> for MaliciousContext<'a, F> {
     }
 }
 
+/// Sometimes it is required to reinterpret malicious context as semi-honest. Ideally
+/// protocols should be generic over `SecretShare` trait and not requiring this cast and taking
+/// `ProtocolContext<'a, S: SecretShare<F>, F: Field>` as the context. If that is not possible,
+/// this implementation makes it easier to reinterpret the context as semi-honest.
+impl<'a, F: Field> AccessSemiHonestContextFromMalicious<'a, F> for MaliciousContext<'a, F> {
+    fn accumulate_macs(self, record_id: RecordId, x: &MaliciousReplicated<F>) {
```

Comment:
What does this function have to do with "access semi honest context from malicious"? This accumulator isn't present for Semi-Honest contexts?

Reply:
I'll see if I can find a better name.  This is really just all the stuff that we need for internal purposes.